agent/php_wrapper.c

```diff
@@ -52,12 +48,53 @@ nruserfn_t* nr_php_wrap_user_function_before_after_clean_with_transience(
                      "for %.*s, but "
                      "it is already set",
                      __func__, NRSAFELEN(namelen), NRBLANKSTR(name));
-    return wraprec;
+    return;
   }
 
   wraprec->special_instrumentation = after_callback;
   wraprec->special_instrumentation_before = before_callback;
   wraprec->special_instrumentation_clean = clean_callback;
+}
+
+nruserfn_t* nr_php_wrap_user_function_before_after_clean_with_transience(
+    const char* name,
+    size_t namelen,
+    nrspecialfn_t before_callback,
+    nrspecialfn_t after_callback,
+    nrspecialfn_t clean_callback,
+    nr_transience_t transience) {
+
+  nruserfn_t* wraprec
+      = nr_php_add_custom_tracer_named(name, namelen, transience);
+
+  nr_php_wraprec_add_before_after_clean_callbacks(name, namelen, wraprec,
+                                                  before_callback,
+                                                  after_callback,
+                                                  clean_callback);
+
+  return wraprec;
+}
+
+nruserfn_t* nr_php_wrap_callable_before_after_clean(
+    zend_function* callable,
+    nrspecialfn_t before_callback,
+    nrspecialfn_t after_callback,
+    nrspecialfn_t clean_callback) {
+  char* name = NULL;
+
+  /* creates a transient wraprec */
+  nruserfn_t* wraprec = nr_php_add_custom_tracer_callable(callable TSRMLS_CC);
+
+  /*
+   * For logging purposes, let's create a name if we're logging at verbosedebug.
+   */
+  if (nrl_should_print(NRL_VERBOSEDEBUG, NRL_INSTRUMENT)) {
+    name = nr_php_function_debug_name(callable);
+  }
+  nr_php_wraprec_add_before_after_clean_callbacks(name, nr_strlen(name), wraprec,
+                                                  before_callback,
+                                                  after_callback,
+                                                  clean_callback);
```

Comment:
According to [this comment](https://github.com/newrelic/newrelic-php-agent/blob/main/agent/php_agent.h#L585-L586) the `name` should be freed if it got created:
```suggestion
                                                  clean_callback);
  if (nrl_should_print(NRL_VERBOSEDEBUG, NRL_INSTRUMENT) && NULL != name) {
    nr_free(name);
  }
```